# Customer Purchase Prediction using Decision Trees

**Predicting buying behavior from a synthetic but realistic dataset:**
- Defining the problem statement
- Collecting or importing the data
- Exploring the dataset and handling the errors in data
- Exploratory data analysis
- Feature engineering
- Modeling
- Testing

## 1. Defining the problem statement

We need to analyze what sort of people were likely to purchase a product.

In addition, we need to use decision tree algorithm in order to predict which individuals are going to puchased based their data.

## 2. Data collection

This dataset is generated by ChatGPT (GPT-5) in order to perform machine learning to tools to predict customer behavior.

Lets import the data.

In [1]:
# Importing  required library

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('data/customer_purchase_dataset.csv')
df.head()

Age_Group  Gender Income_Level   Car_Type Marital_Status Has_Children  \
0     46-60  Female       Medium      Sedan         Single          Yes   
1       60+  Female       Medium      Truck       Divorced          Yes   
2     36-45    Male       Medium  Hatchback        Married          Yes   
3       60+    Male         High      Truck         Single          Yes   
4       60+  Female       Medium      Truck        Married           No   

     Education Credit_Score_Category Purchased  
0     Bachelor                  Good        No  
1     Bachelor                  Good        No  
2     Bachelor                  Good        No  
3  High School             Excellent        No  
4     Bachelor                  Good        No

## 3. Exploring the dataset and handling the errors in data

In [3]:
df.shape

(2000, 9)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Age_Group              2000 non-null   object
 1   Gender                 2000 non-null   object
 2   Income_Level           2000 non-null   object
 3   Car_Type               2000 non-null   object
 4   Marital_Status         2000 non-null   object
 5   Has_Children           2000 non-null   object
 6   Education              2000 non-null   object
 7   Credit_Score_Category  2000 non-null   object
 8   Purchased              2000 non-null   object
dtypes: object(9)
memory usage: 140.8+ KB


In [9]:
# Finding the numbers of duplicated rows
df.duplicated().sum()

np.int64(139)

In [12]:
# Dropping the duplicates 
df.drop_duplicates(inplace=True)

In [13]:
df.shape

(1861, 9)

In [14]:
# Checking the numbe rof null values
df.isna().sum()

Age_Group                0
Gender                   0
Income_Level             0
Car_Type                 0
Marital_Status           0
Has_Children             0
Education                0
Credit_Score_Category    0
Purchased                0
dtype: int64

⭐ **Conclution:**
- There were 2000 values and 9 features in this dataset.
- There were 139 duplicated rows, which were dropped.
- There is no missing value.
- Final number of values is 1861.